src/BTVNanoCommissioning/utils/correction.py

```diff
@@ -3,23 +3,152 @@
 import pickle
 import contextlib
 import cloudpickle
-
+import os
 import numpy as np
 import awkward as ak
-from coffea.lookup_tools.lookup_base import lookup_base
 from coffea.lookup_tools import extractor
 
-from coffea.analysis_tools import Weights
-
 from coffea.lumi_tools import LumiMask
 from coffea.btag_tools import BTagScaleFactor
 from coffea.jetmet_tools import JECStack, CorrectedJetsFactory
 from coffea.lookup_tools import extractor
+import correctionlib
+from BTVNanoCommissioning.utils.AK4_parameters import correction_config
+from BTVNanoCommissioning.helpers.cTagSFReader import getSF
+
+
+def load_SF(campaign):
+    correction_map = {}
+    for SF in correction_config[campaign].keys():
+        if SF == "JME" or SF == "lumiMask":
+            continue
+        if SF == "PU":
+            ## load from correction config
+            if os.path.exists(
+                f"src/BTVNanoCommissioning/jsonpog-integration/POG/LUM/{campaign}"
+            ):
+                correction_map["PU"] = correctionlib.CorrectionSet.from_file(
+                    f"src/BTVNanoCommissioning/jsonpog-integration/POG/LUM/{campaign}/puWeights.json.gz"
+                )
+            ##
+            else:
+                _pu_path = f"BTVNanoCommissioning.data.PU.{campaign}"
+                with importlib.resources.path(
+                    _pu_path, correction_config[campaign]["PU"]
+                ) as filename:
+                    if str(filename).endswith(".pkl.gz"):
+                        with gzip.open(filename) as fin:
+                            correction_map["PU"] = cloudpickle.load(fin)[
+                                "2017_pileupweight"
+                            ]
+                    elif str(filename).endswith(".histo.root"):
+                        ext = extractor()
+                        ext.add_weight_sets([f"* * {filename}"])
+                        ext.finalize()
+                        correction_map["PU"] = ext.make_evaluator()["PU"]
```

Comment:
ah I think I now see the reason for specifying `None` for PU in `AK4_parameters.py`: as they do not exist for Run 3 (in fact, only for Run 2 UL), the final choice whether to use correctionlib or the maintainer pickle files is done here. Might be worth explaining this with a remark/note in the readme

Reply:
Indeed specify `None` is taking the info from json-integration for PU as explained [above](https://github.com/cms-btv-pog/BTVNanoCommissioning/pull/50#discussion_r1094750619). 

The new correction class try to check whether the corrections are exist in jsonpog-integration directory, if not use the custom tagger

I would put info in readme on the correction where `.root` and `pkl` still allow to use.